pkg/types/credentials.go

```diff
@@ -32,16 +32,16 @@ type Basic struct {
 }
 
 type TLS struct {
-	Certificate string `json:"client_certificate,omitempty"`
-	Key         string `json:"client_key,omitempty"`
+	Certificate      string `json:"client_certificate,omitempty"`
+	Key              string `json:"client_key,omitempty"`
+	UseSMCertificate bool   `json:"sm_provided_tls_credentials,omitempty"`
```

Comment:
the naming here is a bit odd, why not leave it SMProvidedTLSCredentials?
later in broker ToObject you use 'tls := &types.TLS{UseSMCertificate: e.SMProvidedCredentials}', so you need to match the strings, either change in the broker object or keep this one as is. 

WDYT?

Reply:
Eyal  told us  to use UseSMCertificate after we already agreed on API with xrs.  What do you think is better to use "useSMCerificate" or "SMProvidedTLSCredentials" i will align

Reply:
please change it to SMProvidedTLSCredentials it's confusing.